packages/backend/src/accounting/psql/service.ts

```diff
@@ -211,9 +243,66 @@ export async function createTransfer(
 
 async function createAccountDeposit(
   deps: ServiceDependencies,
-  { id, account, amount }: Deposit
+  args: Deposit
 ): Promise<void | TransferError> {
-  throw new Error('Not implemented')
+  const {
+    id: transferRef,
+    account: {
+      id: accountRef,
+      asset: { id: assetRef }
+    },
+    amount
+  } = args
+
+  const [account, settlementAccount] = await Promise.all([
+    deps.ledgerAccountService.getLiquidityAccount(accountRef),
+    deps.ledgerAccountService.getSettlementAccount(assetRef)
+  ])
+
+  if (!account) {
+    return TransferError.UnknownDestinationAccount
+  }
+
+  if (!settlementAccount) {
+    return TransferError.UnknownSourceAccount
+  }
+
+  const transferError = validateTransfer({
+    amount,
+    creditAccount: account,
+    debitAccount: settlementAccount
+  })
+
+  if (transferError) {
+    return transferError
+  }
+
+  const transfer: CreateTransferArgs = {
+    transferRef,
+    creditAccountId: account.id,
+    debitAccountId: settlementAccount.id,
+    amount,
+    ledger: settlementAccount.ledger,
+    type: LedgerTransferType.DEPOSIT,
+    state: LedgerTransferState.POSTED
+  }
+
+  try {
+    await deps.knex.transaction(async (trx) => {
+      await deps.ledgerTransferService.createTransfers([transfer], trx)
+    })
+  } catch (error) {
+    if (error instanceof UniqueViolationError) {
+      return TransferError.TransferExists
+    }
+
+    deps.logger.error(
+      { errorMessage: error && error['message'], transferRef, accountRef },
+      'Could not create deposit'
+    )
+
+    throw error
+  }
```

Comment:
It feels weird to not have to do any balance checks here. In tigerbeetle speak, the settlement (debit) account that ensures `credits_must_not_exceed_debits` is being debited, so the 
```
account.credits_pending + account.credits_posted + transfer.amount > account.debits_posted
```
check does not apply. Similarily, the liquidity (credit) account that ensures `debits_must_not_exceed_credits`, is being credited, so the
```
account.debits_pending + account.debits_posted + transfer.amount > account.credits_posted
``` 
check does not apply.


For now, are we just assuming the settlement account always has enough $$$ to fund deposits?

Reply:
I also might move more of this into `ledger-transfers/index.ts` once I implement more transfer methods

Reply:
what if we did check balances (in `createTransfers`)?

Reply:
Yep, we can do that, I was just trying to see once I have the other methods in what could stay in the service itself, vs go into `createTransfers`.

For this case with deposits, is
> For now, are we just assuming the settlement account always has enough $$$ to fund deposits?

the right assumption?

Reply:
correct

Reply:
Balance checks moved to: https://github.com/interledger/rafiki/pull/1111/files#diff-66a9f0afadd8190ee2c83380f68156a747f881b15ec41c47dda14c229eeb26adR143-R156